parla/task_runtime.py

```diff
@@ -1490,52 +1503,50 @@ def _assignment_policy(self, task: Task):
             nonlocal_data /= device.resources['memory']
 
             # Figure out whether the task has a dependency running on this device
-            this_device_owns_dependency = False
+            this_device_owns_dependency = 0
             for dependency in task.dependencies:
                 if device in dependency.req.devices:
-                    this_device_owns_dependency = True
+                    this_device_owns_dependency = 1
                     break
 
-            # If the best device owns a dependency, but this one doesn't,
-            # AND the best device has more local data, we can skip this device
-            # TODO (ses): Ignore this if the dependency has finished running!
-            if best_device_owns_dependency and not this_device_owns_dependency and best_device_local_data >= local_data:
-                continue
-            
             # Next we calculate the load-balancing factor
             # For now this is just a count of tasks on the device queue (TODO (ses): better heuristics later...)
             dev_load = self._device_compute_task_counts[device]
 
             # Normalize this too so we have numbers between 0 and 1
-            dev_load /= self._active_task_count
+            if self._active_compute_task_count > 0:
+                dev_load /= self._active_compute_task_count
+            else:
+                dev_load = 0
 
             # TODO (ses): Move these magic numbers somewhere better
             local_data_weight = 30.0
             nonlocal_data_weight = 10.0
             load_weight = 1.0
+            dependency_weight = 0.5
 
             # Calculate the suitability
             suitability = local_data_weight * local_data \
                         - nonlocal_data_weight * nonlocal_data \
-                        - load_weight * dev_load
+                        - load_weight * dev_load \
+                        + dependency_weight * this_device_owns_dependency
```

Comment:
just a future note I think this should depend on the amount of data shared with the dependency maybe